python/kvikio/_lib/libnvcomp_ll.pyx

```diff
@@ -76,7 +76,7 @@ class nvCompBatchAlgorithm(ABC):
         self,
         size_t batch_size,
         size_t max_uncompressed_chunk_bytes,
-    ) -> (nvcompStatus_t, size_t):
+    ) -> tuple[nvcompStatus_t, size_t]:
```

Comment:
So I guess this is where the dependency on Python 3.9+ is introduced (PEP-585), right? Up until now, `kvikio` did not use any Python feature which would specifically require 3.9+.

Reply:
That's an interesting point I didn't consider. I believe in this particular context this change is actually backwards compatible because using tuple directly has been supporting since Python 3.7, it [just wasn't fully featured](https://peps.python.org/pep-0585/#implementation):

> Usefulness of this syntax before [PEP 585](https://peps.python.org/pep-0585) is limited as external tooling like Mypy does not recognize standard collections as generic. Moreover, certain features of typing like type aliases or casting require putting types outside of annotations, in runtime context.

I would just need to add a `from __future__ import annotations` to this file. I don't know if Cython 3.0.0 relies on Python 3.9+ functionality to recognize this type, but it wouldn't matter much anyway here since this function isn't a `cdef` function. If you prefer I could switch to using `typing.tuple` for now instead though. That may eventually be deprecated, but not for a couple of years at least, and it would be fully backwards compatible.

Reply:
Apologies for not responding sooner.  I think adding `from __future__ import annotations` is a good idea. The package already requires 3.9+ anyway, so anyone willing to use it with 3.8 will be aware of potential compatibility issues.